tests/unit/conftest.py

```diff
@@ -20,6 +26,14 @@ def mock_access_token(mock_response):
         yield vmc_access_token
 
 
+@pytest.fixture
+def mock_vcenter_headers(mock_response):
+    with patch("saltext.vmware.utils.vmc_vcenter_request.get_headers") as vcenter_headers:
+        mock_response.json = mock.Mock("mock-header")
+        vcenter_headers.return_value = mock_response
+        yield vcenter_headers
```

Comment:
**suggestion** this fixture is probably less desirable. Currently it's possible to remove both `get_api_session_id` and `get_headers` from `utils/vmc_vcenter_request.py` and all the unit tests pass.

A possible approach would be just to mock `requests.post` to return something like:

```
with patch("requests.post", autospec=True) as fake_request:
    fake_request.json.return_value = {'expected json': 'dictionary goes here'}
    yield
```

Reply:
For vcenter APIs to be working properly, we need to pass api_session_id (based on vcenter_host, username and password) as header. Therefore, we should not remove `get_api_session_id` and `get_headers` from `utils/vmc_vcenter_request.py`